src/wallet3/output_selection/output_selection.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <vector>
+#include "../output.hpp"
+
+namespace wallet
+{
+  // SelectOutputs will choose some a subset of outputs from the provided list of outputs according to the 
+  // output selection algorithm. The sum of the amounts in the returned outputs will be greater than the amount
+  // passed as the second parameter.
+  std::vector<Output> SelectOutputs(const std::vector<Output>& available_outputs, int64_t amount);
```

Comment:
i feel like SelectOutputs would be better not being a free function.
would suggest something like a `class OutputSelector` with `std::vector<Output> operator()(const std::vector<Output>&, int64_t);` myabe.